natlib/svcd.c

```diff
@@ -183,5 +190,38 @@ static int svcd_init( lua_State *L )
         lua_call(L, 3, 0);
     }
 
+    
+    return 0;
+}
+
+// Lua: storm.n.ndispatch ( pay, srcip, srcport )
+// Dispatches an incoming notification
+
+static int svcd_ndispatch( lua_State *L )
+{
+    // Checks to make sure it is passed 3 parameters
+    if (lua_gettop(L) != 3) return luaL_error(L, "Expected (pay, srcip, srcport)"); 
+
+    size_t parlen;
+    const char* pay = lua_tolstring(L, -1, &parlen); 
+    uint16_t ivkid = lua_tonumber(L, -1); // changes pay parameter to number and returns
+    lua_getglobal(L, "SVCD"); // gets table
+    lua_pushstring(L, "oursubs"); // use oursubs as the key
+    lua_gettable(L, 5);
+    char ivkidstr[16];
+    sprintf(ivkidstr, "%d", ivkid);
+    lua_pushstring(L, ivkidstr);
+    lua_gettable(L, 6);
+
```

Comment:
I'd say either way works. this numbering scheme seems to step up - you can see the stack grow from 5 to 7 elements, and no items are popped from the stack so it's pretty simple. 
If you do renumber it to -1, then I'd definitely suggest doing so for the rest of the stack indices, for consistency.
Fantastic Four